src/Windows/Avalonia.Win32/RenderLoop.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Reactive.Disposables;
+using Avalonia.Rendering;
+using Avalonia.Win32.Interop;
+
+namespace Avalonia.Win32
+{
+    internal class RenderLoop : DefaultRenderLoop
+    {
+        private UnmanagedMethods.TimeCallback timerDelegate;
+
+        public RenderLoop(int framesPerSecond)
+            : base(framesPerSecond)
+        {
+        }
+
+        protected override IDisposable StartCore(Action tick)
+        {
+            timerDelegate = (id, uMsg, user, dw1, dw2) => tick();
+
+            var handle = UnmanagedMethods.timeSetEvent(
```

Comment:
Yep, it runs in its own thread, it's the responsibility of whatever subscribes to the render loop to handle passing the notification to the correct thread - I should document that fact.

At the moment we're still going via WM_PAINT to do the rendering so that will move us onto the right thread; later on we'll probably paint on the render thread to an offscreen bitmap and render sections of that on WM_PAINT as needed.